src/test/java/io/lettuce/core/cluster/pubsub/RedisClusterPubSubConnectionIntegrationTests.java

```diff
@@ -126,13 +133,55 @@ void subscribeToShardChannelViaReplica() {
         int clusterKeyslot = connection.sync().clusterKeyslot(shardChannel).intValue();
         String thisNode = connection.getPartitions().getPartitionBySlot(clusterKeyslot).getNodeId();
 
-        RedisPubSubAsyncCommands<String, String> replica =
-                pubSubConnection.async().nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
+        RedisPubSubAsyncCommands<String, String> replica = pubSubConnection.async()
+                .nodes(node -> thisNode.equals(node.getSlaveOf())).commands(0);
         replica.ssubscribe(shardChannel);
 
         Wait.untilEquals(shardChannel, connectionListener.getShardChannels()::poll).waitOrTimeout();
     }
 
+    @Test
+    @EnabledOnCommand("SSUBSCRIBE")
```

Comment:
Not sure if we should have SSUBSCRIBE here.

I assume is is extremely unlikely that the SSUBSCRIBE be supported by the server and the SPUBLISH - not, but if this is a test to test the SPUBLISH it probably has precedence? On the other hand we have no way to verify the SPUBLISH without a SSUBSCRIBE... Never mind, let's leave it like this

Reply:
i agree with your point, i was in between putting both guards and choosing one of them.